vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueType
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueType Kind;
+    UInt64 Value;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
+  {
+  };
+
+  delegate JsiValueData JsiHostFunction(IJsiRuntime runtime, JsiValueData thisValue, JsiValueData[] args);
+
+  [webhosthidden]
+  interface IJsiHostObject
+  {
+    JsiValueData GetProperty(IJsiRuntime runtime, UInt64 name);
+    void SetProperty(IJsiRuntime runtime, UInt64 name, JsiValueData value);
+    IVectorView<UInt64> GetPropertyNames(IJsiRuntime runtime);
+  };
+
+  [webhosthidden]
+  interface IJsiRuntime
+  {
+    JsiValueData EvaluateJavaScript(IJsiBuffer buffer, String sourceUrl);
+    JsiPreparedJavaScript PrepareJavaScript(IJsiBuffer buffer, String sourceUrl);
+    JsiValueData EvaluatePreparedJavaScript(JsiPreparedJavaScript js);
+    UInt64 Global { get; };
+    String Description { get; };
+    Boolean IsInspectable { get; };
+
+    UInt64 CloneSymbol(UInt64 symbol);
+    UInt64 CloneString(UInt64 str);
+    UInt64 CloneObject(UInt64 obj);
+    UInt64 ClonePropertyNameId(UInt64 propertyNameId);
+
+    UInt64 CreatePropertyNameIdFromAscii(UInt8[] ascii);
```

Comment:
Why not use string?

Reply:
I would like to reduce conversions between one-byte to two-byte characters. Having the byte array seems to be a the best approach to achieve it. We can add a method to simplify creation for cases when we start with an UTF-16 string.

---
In reply to: [451856665](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451856665) [](ancestors = 451856665)